src/ray/gcs/redis_module/ray_redis_module.cc

```diff
@@ -675,16 +675,35 @@ std::string DebugString() {
 
 int DebugString_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
   REDISMODULE_NOT_USED(argv);
-  RedisModule_AutoMemory(ctx);
 
   if (argc != 1) {
     return RedisModule_WrongArity(ctx);
   }
   std::string debug_string = DebugString();
   return RedisModule_ReplyWithStringBuffer(ctx, debug_string.data(), debug_string.size());
 }
+};
+
+namespace auto_memory_redis_commands {
+// Wrap all Redis commands with Redis' auto memory management.
+AUTO_MEMORY(TableAdd_RedisCommand);
+AUTO_MEMORY(TableAppend_RedisCommand);
+AUTO_MEMORY(TableLookup_RedisCommand);
+AUTO_MEMORY(TableRequestNotifications_RedisCommand);
+AUTO_MEMORY(TableCancelNotifications_RedisCommand);
+AUTO_MEMORY(TableTestAndUpdate_RedisCommand);
+AUTO_MEMORY(DebugString_RedisCommand);
+#if RAY_USE_NEW_GCS
+AUTO_MEMORY(ChainTableAdd_RedisCommand);
+AUTO_MEMORY(ChainTableAppend_RedisCommand);
+#endif
+};
 
 extern "C" {
+// Only use commands that have auto memory management.
+// TODO(swang): Ideally, we would make the commands that don't have auto memory
+// management inaccessible instead of just using separate namespaces.
+using namespace auto_memory_redis_commands;
```

Comment:
how about we get rid of this, just have the redis_commands namespace that hides the non-auto ones, and put the others into the global namespace? That would be a bit simpler and would make the auto ones more "canonical".

Reply:
we could even call the redis_command namespace something else, like internal etc. to make clear it is "hidden"

Reply:
Ah good idea, thanks! Now I'm not sure why I started with two separate namespaces :)